File: solutions/7/7.js
function nthPrime(position) {
    let max_size = 1000000;

    let is_prime = Array(max_size).fill(1);

    let i, j, prime_position = 0;
    for (i = 2; i < max_size; i++) {
        if (is_prime[i] == 1) {
            prime_position += 1;

            if (prime_position == position) {
                return i;
            }

            for (j = i * 2; j < max_size; j += i) {
                is_prime[j] = 0;
            }
        }
    }

    return -1;
}

function main() {
    result = nthPrime(10001);

    console.log(result);
}

main();

File: solutions/7/7.py
def nthPrime(position):
    max_size = 1000000

    is_prime = [1] * max_size

    prime_position = 0
    for i in range(2, max_size):
        if (is_prime[i] == 1):
            prime_position += 1

            if (prime_position == position):
                return i

            for j in range(i * 2, max_size, i):
                is_prime[j] = 0
    return -1

def main():
    result = nthPrime(10001)
    print(result)

if __name__ == "__main__":
    main()
